### DISCUSSION TITLE
Enrolled events are not coming in user enrolment list API

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Lern/Community#198] [Issues]
I hope this message finds you well. We are reaching out to request assistance in implementing events functionality on our Sunbird ED portal.

Details of the custom fields required are as follows

1. We have enrolled Event APIs and created object type for event
2. We have tried creating an Event and add batch for it.
3. We enrolled the same event with  ../learner/course/v1/enrol API 
4. I tried following API to get enrolled events but i can see only enrolled courses in response

   
 ```
 curl --location 'host/learner/course/v1/user/enrollment/list/userid' \
--header 'Accept: */*' \
--header 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
--header 'Connection: keep-alive' \
--header 'Cookie: user_cookie' \
--header 'If-None-Match: W/"ee0-OYrb9WlNptVTknnb8vQ4GiTJ6vc"' \
--header 'Referer: Host/events' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
--header 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "Linux"'
```

Where Can I get list of enrolled events list ? What might have missed ?

 @rhwarrier, @komalm, @manishakapadnis, @maheshkumargangula @amit-tarento

**Comment by mahajanmahesh935** [comment]
Hello Team,

We are facing an issue retrieving the enrolled events list for users. When we call the `host/learner/course/v1/user/enrollment/list/userid` API, we are only receiving enrolled courses, not events. Can you please guide us on how to resolve this issue?

**Comment by paritshivani** [comment]
Do we need to do any configuration for this?
where can i get enrolled events? please guide us here

**Reply by komalm** [reply]
@pallakartheekreddy Can we connect today?
We need urgent guidance as we are blocked on this.

Please let me know the suitable time.

cc: @manishakapadnis @paritshivani @mahajanmahesh935

**Comment by paritshivani** [comment]
@amit-torento, @Hari-stackroute , @AmiableAnil 
Any updates on this?? Its a major blocker for us and effecting on our important feature required by client

**Comment by amitpriyadarshi** [comment]
@paritshivani you need to follow below steps:

- check user-enrolment table and validate user and event entry is there in the table or not.
- by default enrolment list api returns data from redis  cache `{userId}:user-enrolments` (index value you can check from the lms service config value `redis.dbIndex`). Check the redis entry and delete from cahce.
- again call enrolment list api, it should fetch the value (if data exists in above mentioned table and redis entry is deleted from respective index).

For reference: In class: https://github.com/Sunbird-Lern/lms-service/blob/release-5.4.0/course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala
refer metod `list`

```val response = if (isCacheEnabled && request.getContext.get("cache").asInstanceOf[Boolean])
                getCachedEnrolmentList(userId, () => getEnrolmentList(request, userId, courseIdList)) else getEnrolmentList(request, userId, courseIdList)
            sender().tell(response, self)```

cc: @AmiableAnil

**Reply by paritshivani** [reply]
@amitpriyadarshi , @AmiableAnil , 
As per the solution given by you,

- We confirmed that the enrolment table has entry for the Event.
- We also tried deleting Radis cache as specified in above response and tried hitting list API again. 
- We are still not able to get the event data in list API.
- We observed that it is not coming in Redis cache also. 

**Where we can check to resolve this caching issue ???**
